src/stuf/stuf_zds/tests/test_client.py

```diff
@@ -0,0 +1,142 @@
+from django.test import TestCase
+
+import requests_mock
+from zgw_consumers.constants import CertificateTypes
+from zgw_consumers.models import Certificate
+
+from ...tests.factories import StufServiceFactory
+from ..client import StufZDSClient
+
+TEST_CERTIFICATES = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")
+TEST_XML = os.path.join(
+    settings.BASE_DIR,
+    "src",
+    "stuf",
+    "stuf_zds",
+    "templates",
+    "stuf_zds",
+    "soap",
+)
+
+
+@requests_mock.Mocker()
+class StufZdsClientTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        certificate_f = open(os.path.join(TEST_CERTIFICATES, "test.certificate"), "r")
+        key_f = open(os.path.join(TEST_CERTIFICATES, "test.key"), "r")
+
+        cls.client_certificate = Certificate.objects.create(
+            label="Test client certificate",
+            type=CertificateTypes.key_pair,
+            public_certificate=File(certificate_f, name="test.certificate"),
+            private_key=File(key_f, name="test.key"),
+        )
+        cls.client_certificate_only = Certificate.objects.create(
+            label="Test client certificate (only cert)",
+            type=CertificateTypes.cert_only,
+            public_certificate=File(certificate_f, name="test1.certificate"),
+        )
+        cls.server_certificate = Certificate.objects.create(
+            label="Test server certificate",
+            type=CertificateTypes.cert_only,
+            public_certificate=File(certificate_f, name="test2.certificate"),
+        )
+
+        certificate_f.close()
+        key_f.close()
+
+        cls.client_options = {
+            "gemeentecode": "1234",
+            "omschrijving": "my-form",
+            "zds_zaaktype_code": "zt-code",
+            "zds_zaaktype_omschrijving": "zt-omschrijving",
+            "zds_zaaktype_status_code": "zt-st-code",
+            "zds_zaaktype_status_omschrijving": "zt-st-omschrijving",
+            "zds_documenttype_omschrijving_inzending": "dt-omschrijving",
+            "referentienummer": "123-123-123",
+        }
+
+    def test_mutual_tls(self, m):
+        stuf_service = StufServiceFactory.create(
+            soap_service__client_certificate=self.client_certificate,
+            soap_service__server_certificate=self.server_certificate,
+        )
+
+        with open(os.path.join(TEST_XML, "creeerZaak.xml"), "r") as f:
+            m.post(
+                stuf_service.soap_service.url,
+                content=Template(f.read()).render(Context({})).encode("utf-8"),
+            )
+
+        client = StufZDSClient(stuf_service, self.client_options)
+
+        client.create_zaak(zaak_identificatie="ZAAK-01", zaak_data={}, extra_data={})
+
+        history = m.request_history
+        request_with_tls = history[-1]
+
+        self.assertEqual(
+            self.server_certificate.public_certificate.path, request_with_tls.verify
+        )
+        self.assertTupleEqual(
```

Comment:
please use the bare `self.assertEqual` - this will call the type-specific method, see https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual